driver/mounter.go

```diff
@@ -214,47 +230,41 @@ func (m *mounter) Unmount(target string) error {
 	return mount.CleanupMountPoint(target, m.kMounter, true)
 }
 
-func (m *mounter) IsRunning(source string, cmdContext execContext) bool {
-	cmd := fmt.Sprintf("ls -l %s | awk '{print $NF}'", source)
-	out, err := cmdContext("/bin/sh", "-c", cmd, source).CombinedOutput()
-
+func (m *mounter) IsRunning(av iAttachmentValidator, source string) bool {
+	out, err := av.evalSymlinks(source)
 	if err != nil {
 		m.log.WithFields(logrus.Fields{
-			"cmd": cmd,
-		}).Errorf("error retrieving the mount name, err: %v", err)
+			"source": source,
+		}).Errorf("error evaluating the sym link, err: %v", err)
 
 		return false
 	}
 
-	mountName := strings.TrimSuffix(string(bytes.Replace(out, []byte("../../"), []byte(""), 1)), "\n")
-
-	cmd = fmt.Sprintf("cat /sys/class/block/%s/device/state | grep -x running | wc -l", mountName)
-	out, err = cmdContext("/bin/sh", "-c", cmd).Output()
+	_, file := filepath.Split(out)
+	fileContent, err := av.readFile(fmt.Sprintf("/sys/class/block/%s/device/state", file))
 	if err != nil {
 		m.log.WithFields(logrus.Fields{
-			"cmd":       cmd,
-			"mountName": mountName,
-		}).Errorf("error retrieving the running state, err: %v", err)
+			"source": source,
+			"file":   file,
+		}).Errorf("error reading the device state file, err: %v", err)
 
 		return false
 	}
 
-	if strings.TrimSuffix(string(out), "\n") != "1" {
+	m.log.WithFields(logrus.Fields{
+		"source":      source,
+		"fileContent": string(fileContent),
+	}).Info("state file content")
+
+	if string(fileContent) != fmt.Sprintf("%s\n", runningState) {
```

Comment:
Let's check DO-internally if this is the best way to check for the running state.